src/psyclone/dynamo0p3.py

```diff
@@ -7621,8 +7648,19 @@ def start_expr(self):
             if loop is self:
```

Comment:
We can't, indeed as yuo said it's because of `__eq__`: if you have a domain loop in the schedule, this 'loop' does not have a variable defined, and it fails in loop.py#113:
```
        is_eq = is_eq and self.variable == other.variable
```
Reason is that the variable getter checks if variable is not None.

IMHO this seems like a bug and should be fixed in loop.py, but since it has nothing to do with this patch, I won't touch it for now (it would affect more than just this one line, there are other accesses to self.variable which would all fail in case of a domain loop).

Reply:
It is mostly correct, except when there is a domain kernel (which will cause a crash when comparing the loops, since it accesses loop.variable and checks if it is defined).

IMHO this should be fixed elsewhere, am happy to have a #TODO here, but I am not sure about the expected functionality of domain loops :(